### PR TITLE
TEG's circulators can now be controlled

### PR DESCRIPTION
**Pull request by AndriiYukhymchak**
Allows control over TEG's circulator inlet and outlet directions via multitool.
Updates generator to accommodate changes and added code that will allow generator to work if TEG's circulators can be made in N/S configuration (hopefully someone can help me with that).

:cl:
tweak: multitool can now be used to swap directions of TEG's circulator
/:cl:

first commit in this repo

### CODE REVIEW

**Comment by Markolie** [comment]
Mhm, I'm pretty sure this was already possible with a multitool, wasn't it?

**Comment by AndriiYukhymchak** [comment]
You are able to swap cold and hot loops via mtool.
This is different - here you swap inlet and outlet sides, so you don't have to have intake always from South side

**Comment by Markolie** [comment]
Ahh, gotcha.

**Comment by AndriiYukhymchak** [comment]
@Markolie can you help me with circulators? for some reason I can't give them back ability to be north/south sided (i.e. vertical TEG)

**Comment by Markolie** [comment]
The TEG was never really intended to be put in a vertical position. I don't think the sprites even support it.

**Comment by AndriiYukhymchak** [comment]
@Markolie it used same sprite for every direction, just rotated. Can you help me understand how to add those positions? When I try to do it, it always wrenches in 1 direction...

**Comment by Markolie** [comment]
You probably need to add or change a variable to the machine that holds the north/south direction, which determines where to wrench from. Not sure if going with the inverse variable's going to do the trick.

**Comment by AndriiYukhymchak** [comment]
I'm talking from building (construction.dm) point of view. Other stuff I can code, but I have 0 idea how to modify its construction

**Comment by AndriiYukhymchak** [comment]
Can this be merged? or we are waiting on something?

**Comment by Markolie** [comment]
construction.dm is just a huge mess. I'd recommend checking for the define that holds the circulators and CTRL+F to look for the change_dir proc or whatever it was called. There's some snowflake code there to handle their rotation.

File: code/ATMOSPHERICS/components/binary_devices/circulator.dm [hunk 1]
```diff
@@ -71,6 +74,43 @@
...
+
+/obj/machinery/atmospherics/binary/circulator/proc/get_outlet_side()
+	if(dir==SOUTH||dir==NORTH)
+		if(sides_inverted==0)
+			return "North"
+		else
+			return "South"
+
+/obj/machinery/atmospherics/binary/circulator/attackby(obj/item/weapon/W as obj, mob/user as mob, params)
+	if(istype(W, /obj/item/device/multitool))
```
Comment (Fox-McCloud) [inline code/ATMOSPHERICS/components/binary_devices/circulator.dm]: Use the istype helper.

File: code/ATMOSPHERICS/components/binary_devices/circulator.dm [hunk 2]
```diff
@@ -71,6 +74,43 @@
 	..()
 	update_icon()
 
+/obj/machinery/atmospherics/binary/circulator/proc/get_inlet_air()
+	if(sides_inverted==0)
```
Comment (Fox-McCloud) [inline code/ATMOSPHERICS/components/binary_devices/circulator.dm]: `if(side_inverted == 0)`

Look out for this jamming together throughout the rest of your PR.

File: code/ATMOSPHERICS/components/binary_devices/circulator.dm (hunk 1 above)
Comment (AndriiYukhymchak) [inline code/ATMOSPHERICS/components/binary_devices/circulator.dm]: what you mean by this?

Comment (Fox-McCloud) [inline code/ATMOSPHERICS/components/binary_devices/circulator.dm]: `if(ismultitool(W))`

**Comment by AndriiYukhymchak** [comment]
@Fox-McCloud Changes made

**Pull request merged**